discord/message_handlers.go

```diff
@@ -42,6 +42,179 @@ func (guild *GuildState) handleGameStartMessage(s *discordgo.Session, m *discord
+
+			var discriminationNecessary = false;
+
+			for _, uName := range idUsernameMap {
+				if uName == username {
+					log.Print("Found a previous user with the same name as current user: " + username);
+					log.Print("Setting discrimination necessary to true");
+					//Came across exact same name. Update all with the same name to also include the discriminator for the name
+					discriminationNecessary = true
+					break;
+				}
+			}
+
+			idUsernameMap[userID] = username;
+
+			if discriminationNecessary {
+				log.Print("Determined that discrimination necessary is true");
+				usernameIdCopy := make(map[string]string);
+
+				log.Print("Looking through the map for similar values.");
+				for uID, uName := range idUsernameMap {
+					var value = uName;
+					if uName == username {
+						log.Print("Found a similar value...");
+						// Is a duplicate/original of the name. Add descriminator
+						var targetMember, _ = s.State.Member(guildId, uID)
+						value = uName + "#" + targetMember.User.Discriminator;
+						log.Print("Updating name to use discriminator: " + targetMember.User.Discriminator);
+					}
+
+					usernameIdCopy[uID] = value;
+				}
+				log.Print("Finished iteration through map for similar values.");
+
+				idUsernameMap = usernameIdCopy;
+
+				log.Print("Updated map");
+
+
+			}
+		} else {
+			log.Print("User has been saved before: " + user.Username);
+			// User has been saved before
+		}
+	}
+
+	return idUsernameMap
+}
+
+func getGuildById(s *discordgo.Session, guildId string) *discordgo.Guild{
+	var g,_ = s.State.Guild(guildId);
+	return g;
+}
+
+func isVCTracked(voiceChannelID string, channels []TrackingChannel) bool {
+	for _, a := range channels {
+		if a.channelID == voiceChannelID {
+			return true;
+		}
+	}
+	return false;
+}
+
+func getMemberById(s *discordgo.Session, userId string, guildId string) *discordgo.Member {
+	log.Print("UserId: " + userId);
+	var member, err = s.State.Member(guildId, userId)
+	if err != nil {
+		// If we were unable to get member using local/cache information, grab it from the discord API
+		// This fallback is much slower, but guaranteed if the first method doesn't work.
+		log.Println("Error: " + err.Error());
+		log.Println("Falling back to s.GuildMember implementation");
+		member, _ = s.GuildMember(guildId, userId);
+	}
+
+	if member == nil {
+		// If member is still nil for some reason, show an error message;
+		log.Print("Member is nil for given user id: " + userId);
+		return nil
+	}
+
+	return member
+}
+
+func getUsername(member *discordgo.Member) string {
+	//Assign username to their nick if they have one, otherwise use their username.
+	var username = "";
+	if member.Nick == "" {
+		// Doesn't have a Nick.
+		username = member.User.Username;
+	} else {
+		// Does have a nickname. Use nickname for reference
+		username = member.Nick;
+	}
+	return username;
+}
+
+func createMessage(s *discordgo.Session, guildState *GuildState) {
```

Comment:
Rename? This name is really generic for what is, in essence, a pretty specific function

Reply:
This is what I've changed it to:
```createInitialMessageAndUpdatePrintedUsers```
I'm not sure if it's too long at this point

Reply:
Imo that's fine; I'm a big advocate for longer function names that clearly indicate the function/purpose, as opposed to preferring brevity at the loss of communication/documentation